pkg/daemon/osrelease/osrelease.go

```diff
@@ -25,17 +37,79 @@ const (
 // OperatingSystem is a wrapper around a subset of the os-release fields
 // and also tracks whether ostree is in use.
 type OperatingSystem struct {
-	// id is the ID field from the os-release
+	// id is the ID field from the os-release or inferred from the OS image
+	// label.
 	id string
-	// variantID is the VARIANT_ID field from the os-release
+	// variantID is the VARIANT_ID field from the os-release or inferred from the
+	// OS image label.
 	variantID string
-	// version is the VERSION, RHEL_VERSION, or VERSION_ID field from the os-release
+	// version is the VERSION, RHEL_VERSION, or VERSION_ID field from the
+	// os-release or image label.
 	version string
-	// osrelease is the underlying struct from github.com/ashcrow/osrelease
-	osrelease osrelease.OSRelease
+	// values is a map of all the values we uncovered either via the
+	// /etc/os-release / /usr/lib/os-release files *or* the labels attached
+	// to an OS image.
+	values map[string]string
+	// source identifies whether this came from the OSRelease file or from image labels.
+	source InfoSource
+}
+
+func newOperatingSystemFromImageLabels(imageLabels map[string]string) (OperatingSystem, error) {
+	os := OperatingSystem{
+		values: imageLabels,
+		source: ImageLabelInfoSource,
+	}
+
+	if isCoreOSVariant(imageLabels) {
+		os.variantID = coreos
+	} else {
+		return os, fmt.Errorf("unable to identify coreos variant from labels: %v", imageLabels)
+	}
+
+	version := ""
+	versionOK := false
+	if version, versionOK = imageLabels["version"]; !versionOK {
+		return os, fmt.Errorf("missing 'version' label: %v", imageLabels)
+	}
+
+	// Only FCOS and SCOS set this field.
+	if osName, osNameOK := imageLabels["io.openshift.build.version-display-names"]; osNameOK {
```

Comment:
It seems odd that only FCOS and SCOS set this field and that RHCOS simply doesn't. It is possible that I examined the wrong container labels when I was setting up the tests for this code.

Reply:
Yeah :cry:   This is because of the [`machine-os-content` zombie issue](https://issues.redhat.com/browse/MCO-392)...some links for this in https://github.com/openshift/os/pull/1048

Basically `oc adm release new` barfs and dies if there are *two*` container images in the release which claim to be an operating system.  OKD already set things up so that `machine-os-content` is an alias, so they don't have this problem.